### PR TITLE
HTTP response compression - fix Undertow and RESTEasy Classsic support

### LINKED ISSUE #31415: GZIP Compression not working in servlet application
### Describe the bug

I am working on the JSF/Faces extensions for Quarkus for OmniFaces and PrimeFaces.

Using `quarkus.http.enable-compression=true` is not compressing any of my JS or CSS files as expected based on Mime Type.

### Expected behavior

GZIP compression should occur

### Actual behavior

No Gzip compression is happening with `quarkus.http.enable-compression=true`

### How to Reproduce?

Reproducer: [quarkus-compression.zip](https://github.com/quarkusio/quarkus/files/10831165/quarkus-compression.zip)

1. Unzip the reproducer
2. Run `mvn quarkus:dev`
3. Open browser to http://localhost:8080/ 
4. Open F12 and inspect the traffic GZIP compression is not taking place even though the `Content-Type: application/javascript` is correct.

image

### Output of `uname -a` or `ver`

Microsoft Windows [Version 10.0.22621.1265]

### Output of `java -version`

OpenJDK Runtime Environment Temurin-17.0.5+8 (build 17.0.5+8)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.16.3

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.6 (84538c9988a25aec085021c365c560670ad80f63)

### Additional information

I have also tested trying to custom set mime-types and it has no effect using `quarkus.http.compress-media-types=` but it also has no effect.  But the defaults of ` text/html, text/plain, text/xml, text/css, text/javascript and application/javascript.` are all I need anyway.

### PR DESCRIPTION
**Pull request by mkouba**
- honor the quarkus.http.compress-media-types in Undertow Servlet and RESTEasy Classsic extensions
- fixes #31415 and #26112

### CODE REVIEW

**Review by geoand — APPROVED** [review]

**Review by melloware — APPROVED** [review]

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/HttpBuildTimeConfig.java [hunk 1]
```diff
@@ -64,14 +64,12 @@ public class HttpBuildTimeConfig {
...
+     * If enabled then the response body is compressed if the {@code Content-Type} header is set and the value is a compressed
+     * media type as configured via {@link #compressMediaTypes}.
      *
-     * Note that this will attempt to compress all responses, to avoid compressing
-     * already compressed content (such as images) you need to set the following header:
-     *
-     * Content-Encoding: identity
-     *
-     * Which will tell vert.x not to compress the response.
+     * Note that the RESTEasy Reactive and Reactive Routes extenions also make it possible to enable/disable compression
```
Comment (gsmet) [inline extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/HttpBuildTimeConfig.java:70]:
```suggestion
     * Note that the RESTEasy Reactive and Reactive Routes extensions also make it possible to enable/disable compression
```

**Review by gsmet — COMMENTED** [review]
Spotted a small typo. Otherwise looks good from what I can see.

**Comment by mkouba** [comment]
@gsmet Do you have an idea why do the `Quarkus CI / Maven Tests` fail?

**Comment by geoand** [comment]
I've seen it happen in an other [PR](https://github.com/quarkusio/quarkus/pull/31353#issuecomment-1441866054) as well, but I have not had time to look into it.

@Sgitario as it is XML related, perhaps you can have a look?

**Comment by Sgitario** [comment]
> I've seen it happen in an other [PR](https://github.com/quarkusio/quarkus/pull/31353#issuecomment-1441866054) as well, but I have not had time to look into it.
> 
> @Sgitario as it is XML related, perhaps you can have a look?

The `quarkus-jaxb` extension is validating the JAXBContext is valid at build time and fails if it's not.
My suspicion is that the `stax-api` dependency brings some `javax.xx` classes that might the JAXB to fail as we now support Jakarta only. 

I have no idea why we use the `stax-api` dependency as part of this test, any hints?

**Comment by geoand** [comment]
> I have no idea why we use the stax-api dependency as part of this test, any hints?

None whatosever :) I have no idea what the test is doing or what it was added.
 
I am pretty sure that we can just remove it and not loose anything WRT test coverage if we can't quickly find a solution.

**Comment by geoand** [comment]
> I have no idea why we use the stax-api dependency as part of this test, any hints?

None whatosever :) I have no idea what the test is doing or what it was added.
 
I am pretty sure that we can just remove it and not loose anything WRT test coverage if we can't quickly find a solution.

**Comment by mkouba** [comment]
The CI failures do not seem to be related. I'll re-run the Devtools Tests - JDK 17 and if it passes I'd like to merge this pr.

**Comment by Sgitario** [comment]
> > I have no idea why we use the stax-api dependency as part of this test, any hints?
> 
> None whatosever :) I have no idea what the test is doing or what it was added.
> 
> I am pretty sure that we can just remove it and not loose anything WRT test coverage if we can't quickly find a solution.

Let's continue the discussion here: https://github.com/quarkusio/quarkus/pull/31494

**Comment by mkouba** [comment]
@gsmet It seems that the Native Tests - Misc4 is flaky, fails in some other PRs as well.

**Pull request merged**